src/collection.rs

```diff
@@ -361,6 +378,38 @@ pub struct CollectionWithId<T> {
     id_to_idx: HashMap<String, Idx<T>>,
 }
 
+/// Creates a `CollectionWithId` from one element.
+///
+/// This cannot fail since there will be a unique identifier in the collection
```

Comment:
you are right, that's why you can use `CollectionWithId::new(vec![object]).unwrap()` without hesitation

Reply:
Done in 1e0da0a40a1a1207612bbd045ae491625908c34f.